src/main/java/com/google/api/codegen/MethodConfig.java

```diff
@@ -147,6 +152,10 @@ public boolean apply(Field input) {
     ImmutableMap<String, String> fieldNamePatterns =
         ImmutableMap.copyOf(methodConfig.getFieldNamePatterns());
 
+    List<String> sampleCodeInitFields = new ArrayList<>();
+    sampleCodeInitFields.addAll(methodConfig.getRequiredFieldsList());
+    sampleCodeInitFields.addAll(methodConfig.getSampleCodeInitFieldsList());
```

Comment:
Your changes didn't cause this, but the methodConfig parameter is confusingly named since we are creating a MethodConfig object, perhaps rename to methodConfigProto?